### ISSUE TITLE
Thanks

### ISSUE THREAD

**Issue by samuel1** [mawww/kakoune#1717]
Hiya, thanks for Kakoune, I'm really impressed with how you designed it

** I'm really new to Kakoune, and I've got a long way to go before I grasp everything, but two immediate thoughts (please feel free to ignore them) are firstly a small one:  ctrl-click adds an extra cursor.  What about something like alt-click to remove an existing cursor?

The second thought is what about using ctrl to access all of normal mode, without leaving insert mode.  Then rather than `<esc>ww;a` or what I really do `<right><right><right><right><right><right><right><right><right>.....` I would just have `<c-w><c-w>`

Then ctrl might need some re-mapping.  shift-click to add cursors rather than ctrl-click should be more fitting with the idea of 'shift to increase selections'.  Then a silly bonus idea would be to allow two key combos, such as `<v-x>` (holding down `v`, and pressing `x` together) -- this would give an very large set of new key combos

Anyway, hope that's slightly helpful, and thanks for a great text editor

**Comment by lenormf** [comment]
Hi,

Can you create separate issues for the capture groups being lost, the pipe/`<c-r>` bug and the crashes? If you compiled the editor with debug symbols, make sure to include a stacktrace in the crash report.

You can edit the first three points out of the post and leave the generic discussion in, otherwise the conversation are going to be hard to follow if several topics are discussed at once.

**Comment by danr** [comment]
Hi,

> The second thought is what about using ctrl to access all of normal mode, without leaving insert mode. 

These are accessible using `<a-;>`, and you could map control bindings in insert mode to go via that. One inspiration how to generate keybindings could be this comment which binds `,k` to `<a-k>` in normal mode for most keys `k`: https://github.com/mawww/kakoune/issues/1016#issuecomment-269575543

Kakoune is very configurable, so just go for you ideas :)

**Comment by samuel1** [comment]
Hiya, just realized today: esc could be swap mode; tap to swap mode, and <esc-any> to do any without leaving the current mode...

**Comment by mawww** [comment]
Thats not really consistent with the rest of Kakoune, there more than 2 modes in Kakoune, so <kbd>esc</kbd> should not switch between modes. Additionally, <kbd>esc</kbd> is already stopping macro recording in normal mode.

**Comment by danr** [comment]
@samuel1 you can make escape behave like that using a tool like https://github.com/alols/xcape 

The downside is that sending the unmodified key has to be done at key up rather than key down, so the experience is not so smooth.
